Source/Diagnostics/ReducedDiags/ColliderRelevant.cpp

```diff
@@ -0,0 +1,572 @@
+                const amrex::Array4<const amrex::Real> & ez_arr = Ez[pti].array();
+                const amrex::Array4<const amrex::Real> & bx_arr = Bx[pti].array();
+                const amrex::Array4<const amrex::Real> & by_arr = By[pti].array();
+                const amrex::Array4<const amrex::Real> & bz_arr = Bz[pti].array();
+                const amrex::IndexType ex_type = Ex[pti].box().ixType();
+                const amrex::IndexType ey_type = Ey[pti].box().ixType();
+                const amrex::IndexType ez_type = Ez[pti].box().ixType();
+                const amrex::IndexType bx_type = Bx[pti].box().ixType();
+                const amrex::IndexType by_type = By[pti].box().ixType();
+                const amrex::IndexType bz_type = Bz[pti].box().ixType();
+
+                // evaluate reduce_op
+                reduce_op.eval(pti.numParticles(), reduce_data,
+                [=] AMREX_GPU_DEVICE (int i) -> ReduceTuple
+                {
+                    // get external fields
+                    amrex::ParticleReal xp, yp, zp;
+                    GetPosition(i, xp, yp, zp);
+                    amrex::ParticleReal ex = 0._rt, ey = 0._rt, ez = 0._rt;
+                    amrex::ParticleReal bx = 0._rt, by = 0._rt, bz = 0._rt;
+                    getExternalEB(i, ex, ey, ez, bx, by, bz);
+
+                    // gather E and B
+                    doGatherShapeN(xp, yp, zp,
+                        ex, ey, ez, bx, by, bz,
+                        ex_arr, ey_arr, ez_arr, bx_arr, by_arr, bz_arr,
+                        ex_type, ey_type, ez_type,
+                        bx_type, by_type, bz_type,
+                        dx_arr, xyzmin_arr, lo,
+                        n_rz_azimuthal_modes, nox, galerkin_interpolation);
+                    // compute chi
+                    amrex::Real chi = 0.0_rt;
+                    if (is_photon) {
+                        chi = QedUtils::chi_photon(ux[i]*m, uy[i]*m, uz[i]*m,
+                                            ex, ey, ez, bx, by, bz);
+                    } else {
+                        chi = QedUtils::chi_ele_pos(ux[i]*m, uy[i]*m, uz[i]*m,
+                                            ex, ey, ez, bx, by, bz);
+                    }
+                    return {chi, chi, chi*w[i]};
+                });
+            }
+            auto val = reduce_data.value();
+            chimin_f = get<0>(val);
+            chimax_f = get<1>(val);
+            chiave_f = get<2>(val);
+            amrex::ParallelDescriptor::ReduceRealMin(chimin_f, amrex::ParallelDescriptor::IOProcessorNumber());
+            amrex::ParallelDescriptor::ReduceRealMax(chimax_f, amrex::ParallelDescriptor::IOProcessorNumber());
+            amrex::ParallelDescriptor::ReduceRealSum(chiave_f, amrex::ParallelDescriptor::IOProcessorNumber());
+
+            m_data[get_idx("chimin_"+species_names[i_s])] = chimin_f;
+            m_data[get_idx("chiave_"+species_names[i_s])] = chiave_f/w_tot;
```

Comment:
This line produces the error `‘w_tot’ was not declared in this scope` when compiled in 1D. I wonder what we should do in 1D anyways. Do these diagnostics make sense? If not, we could simply skip them, as we currently do for RZ. Otherwise, we need to make sure that `w_tot` is defined also in 1D.

Reply:
Also, this should ensure the `w_tot` is not zero.

Reply:
Some of the quantities do make sense in 1D: the derivative of luminosity in time and the `chi` values.
I guess we could go back to the previous version where `w_tot` is computed in a separate reduce operation before the other ones.